### PR TITLE
Only make parse method comparison if it exists

### PR DESCRIPTION
**Pull request by farias-r7**
https://github.com/rapid7/marionette.carpenter/pull/37 added the ability to override the parse method. There was a corner case where it was doing a comparison of an Object {} != null that then caused parse to be called on the null reference. Causing a console error to occur.

```
        if Backbone.Collection.prototype.parse != collection.parse
           collection.parse.apply(this, arguments)
```

We now only do the comparison if collection.parse is defined.

@ugohil-r7 let me know if this is the desired behavior.

The suggested fix is:

```
        if collection.parse? && (Backbone.Collection.prototype.parse != collection.parse)
            collection.parse.apply(this, arguments)
```

# Verification Steps
- [x] Verify travis build passes.

### CODE REVIEW

File: src/entities/paginated_collection.coffee [hunk 1]
```diff
@@ -137,7 +137,7 @@ define [], ->
     #
     # @param data [Object] the response returned from server
       parse: (data) ->
-        if Backbone.Collection.prototype.parse != collection.parse
+        if collection.parse? && (Backbone.Collection.prototype.parse != collection.parse)
```
Comment (ugohil-r7) [inline src/entities/paginated_collection.coffee]: By default backbone collection has parse method defined on it . 

http://backbonejs.org/#Collection-parse

So there is no use of this collection.parse?

**Comment by farias-r7** [comment]
"The default implementation is a no-op"
according to the docs. Should just pass back a JSON response though
I have a feeling it has to do with PaginatedCollection we use
Carpenter takes in a plain Backbone.Collection, but carpenter uses a paginated collection
we copy the object.
2:14 PM
Since parse isn’t defined on the object itself, I don’t think it get’s copied over as we don’t copy the prototype

**Comment by ugohil-r7** [comment]
No Harm in adding null check. Just want to make sure the problem is fix right way.
After looking at code I have a feeling that problem is with this collection "StaticPaginatedCollection" and not with AjaxPaginatedCollection.

coz AjaxPaginatedCollection has parse method and 
StaticPaginatedCollection not sure. 

Can you try to add test case for this scenario if possible. 

:shipit:

**Comment by farias-r7** [comment]
I'll add test coverage. Thanks for the review.

File: src/entities/paginated_collection.coffee (hunk 1 above)
Comment (dcastellanos-r7) [inline src/entities/paginated_collection.coffee]: Does it matter? If parse is there and it is not Backbone.Collection.prototype.parse, then it is false. If collection.parse is null, undefined, or some other falsy value then it is not Backbone.Collection.prototype.parse, and it is false.

**Comment by farias-r7** [comment]
collection.parse? is checking parse property on the collection object which happens to be a Backbone.Collection object and not an instance of a Backbone.Collection. We clone the collection passed into carpenter using .extend({properties}), we don't define parse in the hash as .extend({properties}, {classProperties}).

so parse is defined on the instance and not the Backbone.Collection object. 

Sometimes the user passes in a Backbone.Collection object and other times a instance of Backbone.Collection. The null check covers both.

**Pull request merged**